### PR TITLE
Bump cabal-bootstrap-gen to GHC 9.4.4 ecosystem

### LINKED ISSUE #8613: Bootstrap file for GHC 9.4 on Linux
Apologizes if this is not the right place for this kind of discussion but looking at `bootstrap/` there doesn't seem to be a JSON configuration file for GHC 9.4 on Linux, is there any reason why GHC 9.4 isn't supported out-of-the-box by the bootstrap setup? I am aware that I can create my own bootstrap JSON file for 9.4 by following the instructions in `bootstrap/README.md` I was, however, wondering if there is any official "policy" regarding the addition of these bootstrap files to the Cabal repository.

### PR DESCRIPTION
**Pull request by nmeum**
I updated the dependencies in the .cabal bootstrap file and sightly adjusted src/Main.hs for the `I.riCabal → I.riCabalHash` change. I successfully bootstrapped Cabal on Alpine Linux Edge with these changes. I am new to Cabal so let me know if this needs any adjustments.

Fixes #8613

---

Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: bootstrap/src/Main.hs [hunk 1]
```diff
@@ -93,7 +93,7 @@ main2 meta plan = do
                         return
                             ( Hackage
                             , Just $ fromIntegral (I.riRevision relInfo)
-                            , P.sha256FromByteString $ I.getSHA256 $ I.riCabal relInfo
+                            , P.sha256FromByteString $ I.getSHA256 $ I.riCabalHash relInfo
```
Comment (nmeum) [inline bootstrap/src/Main.hs:96]: This probably needs some sort of `#ifdef` in order to not break bootstrapping with GHC `<9.4.0`.

Comment (Mikolaj) [reply bootstrap/src/Main.hs:96]: @nmeum: thank you! Yes, I guess it does. We'll need to test this with older GHC and tweak as needed.

**Comment by Mikolaj** [comment]
@nmeum: BTW, please feel free to compare with old PRs that update the generator and/or generate files for new GHCs.

File: bootstrap/src/Main.hs (hunk 1 above)
Comment (nmeum) [reply bootstrap/src/Main.hs]: It seems this change was introduced in https://github.com/haskell-CI/haskell-ci/commit/4c28d9961d5d3e69c84205693824cd62fb9a7672 with cabal-install-parsers-0.6. I added a corresponding `#if MIN_VERSION` check.

**Comment by nmeum** [comment]
> @nmeum: BTW, please feel free to compare with old PRs that update the generator and/or generate files for new GHCs.

I tried to follow #8403 but somehow this PR didn't include changes to the top-level `Makefile` and the `CI` I just updated the PR in the hope that I modified these files correctly.

**Review by robx — APPROVED** [review]
This looks good to me, though I haven't tested it locally. Bootstrap CI passes so the generated files should be fine at least. (CI doesn't cover `bootstrap-gen` itself.)

**Review by ulysses4ever — APPROVED** [review]

**Comment by Mikolaj** [comment]
Let me rebase so that CI unbreaks (we worked around some infrastructure breakage).

@nmeum: did you perchance test the new scripts with GHCs other than 9.4.4? Ideally with all listed in the Makefile? I guess it's fine if they produce differenct bootstrap files than in the repo, but a smoke test would be comparing if the sizes are similar.

After your are satisfied about your due diligence, please kindly set the merge_me label.

**Comment by Mikolaj** [comment]
@mergify rebase

**Comment by Mikolaj** [comment]
@mergify rebase

**Comment by nmeum** [comment]
Given that this passes CI and has been approved: Do you need me to do anything else to get this merged or is it just a question of time? :)

**Comment by Kleidukos** [comment]
@nmeum All good on our side, you may set the label `merge-me` and remove the `attention: needs-review` label :)

**Comment by Mikolaj** [comment]
@nmeum: as as @Kleidukos said. And feel free to read CONTRIBUTING.md if in doubt.

BTW, did you catch the below comment of mine?

> @nmeum: did you perchance test the new scripts with GHCs other than 9.4.4? Ideally with all listed in the Makefile? I guess it's fine if they produce differenct bootstrap files than in the repo, but a smoke test would be comparing if the sizes are similar.
> 
> After your are satisfied about your due diligence, please kindly set the merge_me label.

**Comment by nmeum** [comment]
> BTW, did you catch the below comment of mine?
> 
> > @nmeum: did you perchance test the new scripts with GHCs other than 9.4.4? Ideally with all listed in the Makefile? I guess it's fine if they produce differenct bootstrap files than in the repo, but a smoke test would be comparing if the sizes are similar.

Sorry, I somehow missed your comment. I only tested this with GHC 9.4.4. I am using this changeset for bootstrapping Cabal for the [Alpine Linux](https://alpinelinux.org) [Cabal package](https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/community/cabal-stage0) and hence have only tested this with the GHC version available in the Alpine repositories (9.4.4).

**Comment by Mikolaj** [comment]
> I only tested this with GHC 9.4.4. I am using this changeset for bootstrapping Cabal for the [Alpine Linux](https://alpinelinux.org) [Cabal package](https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/community/cabal-stage0) and hence have only tested this with the GHC version available in the Alpine repositories (9.4.4).

That's fine. If somebody manages to test during the 2 days delay period after setting the label, it'd be great. But if not, we still have a long time before cabal 3.10.2 or 3.12, so we may hear from other boostrappers and fix, as necessary.

**Comment by Mikolaj** [comment]
@mergify backport 3.10

**Pull request merged**